corehq/apps/hqcase/api/updates.py

```diff
@@ -275,3 +278,26 @@ def _submit_case_blocks(case_blocks, domain, user, device_id):
         device_id=device_id,
         max_wait=15
     )
+
+
+def validate_update_permission(domain, data, user, is_creation):
```

Comment:
I believe this also needs to consider case indices.  That is, can a restricted user create indices to cases they don't have access to?  I assume the answer should be "no", as that'd give the restricted user the ability to influence cases outside their location.

That might be most easily accomplished by moving the validation inside `handle_case_update` so it can occur on the processed data, and with all IDs populated.

Reply:
That's a good question - you might check with @kaapstorm on that front, as he wrote that quite recently.  However `handle_case_update` is certainly designed as the guts of the API, and it has a decent bit of user-facing validation built in, so it does seem appropriate.  In particular, I'm thinking that it would be tough to do that without access to `JsonIndex.get_id`